src/fheroes2/game/game_scenarioinfo.cpp

```diff
@@ -315,70 +354,66 @@ namespace
                 fheroes2::fadeOutDisplay();
                 break;
             }
-            else if ( le.MouseClickLeft( rectPanel ) ) {
+            else if ( le.MouseClickLeft( roi ) ) {
                 const int32_t index = GetRectIndex( coordDifficulty, le.GetMouseCursor() );
 
                 // select difficulty
                 if ( 0 <= index ) {
-                    RedrawScenarioStaticInfo( rectPanel );
-                    levelCursor.setPosition( coordDifficulty[index].x, coordDifficulty[index].y );
+                    levelCursor.setPosition( coordDifficulty[index].x - levelCursorOffset, coordDifficulty[index].y - levelCursorOffset );
                     levelCursor.redraw();
                     Game::saveDifficulty( index );
-                    RedrawDifficultyInfo( pointDifficultyInfo );
-                    playersInfo.RedrawInfo( false );
-                    ratingRoi = RedrawRatingInfo( rectPanel.getPosition(), rectPanel.width );
-                    buttonOk.draw();
-                    buttonCancel.draw();
-                    display.render();
+                    ratingArea.restore();
+                    ratingRoi = RedrawRatingInfo( roi.getPosition(), roi.width );
+
+                    display.render( background.activeArea() );
```

Comment:
We can simply use `roi` here not to call once again the `activeArea()` method:
```suggestion
                    display.render( roi );
```